ggshield/cmd/hmsl/fingerprint.py

```diff
@@ -119,17 +139,28 @@ def fingerprint_cmd(
 
 
 def collect(
-    input: TextIO, source: SourceType = SourceType.FILE
+    input: TextIO, input_type: InputType = InputType.FILE
 ) -> Iterator[SecretWithKey]:
     """
-    Collect the secrets from the source
+    Collect the secrets
     """
-    for line in input:
-        secret = line.strip()
-        if secret == "":
-            # Skip empty lines
-            continue
-        if source == SourceType.FILE:
+    if input_type == InputType.ENV:
+        config = dotenv_values(stream=input)
+        for key, value in config.items():
+            key: str = str(key)
+            value: str = str(value)
+            # filter our excludes keys and values
+            if key == "" or value == "":
+                continue
```

Comment:
pyright is not happy here because config is of type `dict[str, str|none]` but value is of type `str`. This can be fixed like this:

```suggestion
        for key, value in config.items():
            # filter our excludes keys and values
            if not key or not value:
                continue
```

Using `not` gets rid of both the `value == ""` and `value is None` cases, so pyright knows that value cannot be `None` any more after this.